internal/cephfs/core/metadata.go

```diff
@@ -149,3 +177,22 @@ func (s *subVolumeClient) UnsetAllMetadata(keys []string) error {
 
 	return nil
 }
+
+// ListMetadata returns all the metadata set on the subvolume in a volume.
+// It returns a map of key-value pairs.
+func (s *subVolumeClient) ListMetadata() (map[string]string, error) {
+	if !s.enableMetadata {
+		return nil, nil
+	}
+
+	metadata, err := s.listMetadata()
+	// If listMetadata is not supported return nil
+	if errors.Is(err, ErrSubVolMetadataNotSupported) {
+		return nil, nil
+	}
+	if err != nil && !errors.Is(err, libcephfs.ErrNotExist) {
+		return nil, fmt.Errorf("failed to list metadata on subvolume %v: %w", s, err)
```

Comment:
why do we need extra check for `libcephfs.ErrNotExist` this should return error in anycase not nil when subvolume is not found